src/presentation/ui/assets/additional.js

```diff
@@ -5,24 +5,31 @@ document.addEventListener("alpine:init", () => {
     const loadingOverlayElement = document.getElementById("loading-overlay");
     loadingOverlayElement.classList.add("htmx-request");
 
-    await fetch(url, {
-      method: method,
-      headers: {
-        Accept: "application/json",
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(payload),
-    })
-      .then((response) => {
-        loadingOverlayElement.classList.remove("htmx-request");
-        return response.json();
-      })
-      .then((parsedResponse) => {
-        Alpine.store("toast").displayToast(parsedResponse.body, "success");
-      })
-      .catch((parsedResponse) => {
-        Alpine.store("toast").displayToast(parsedResponse.body, "danger");
-      });
+	try {
+		const response = await fetch(url, {
+			method: method,
+			headers: {
+				Accept: "application/json",
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify(payload),
+		});
+		const parsedResponse = await response.json()
+
+		loadingOverlayElement.classList.remove("htmx-request");
+
+		if (!response.ok) {
+			throw new Error(parsedResponse.body);
+		}
+
+		if (method.toUpperCase() !== "GET") {
+			Alpine.store("toast").displayToast(parsedResponse.body, "success");
+		}
+
+		return parsedResponse.body;
+	} catch (error) {
```

Comment:
I think there is an indentation problem on the try/catch, but are we using the jsonAjax for GETs? Where?